azurerm/internal/services/maintenance/parse/maintenance_assignment_virtual_machine.go

```diff
@@ -0,0 +1,33 @@
+package parse
+
+import (
+	"fmt"
+	"regexp"
+
+	parseCompute "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/parse"
+)
+
+type MaintenanceAssignmentVirtualMachineId struct {
+	VirtualMachineId    *parseCompute.VirtualMachineId
+	VirtualMachineIdRaw string
+	Name                string
+}
+
+func MaintenanceAssignmentVirtualMachineID(input string) (*MaintenanceAssignmentVirtualMachineId, error) {
+	groups := regexp.MustCompile(`^(.+)/providers/Microsoft\.Maintenance/configurationAssignments/([^/]+)$`).FindStringSubmatch(input)
+	if len(groups) != 3 {
+		return nil, fmt.Errorf("parsing Maintenance Assignment Virtual Machine ID: %q: Expected 3 groups", input)
```

Comment:
As in the same func for `dedicated_host`, possibly simplify the message for the user?
```suggestion
		return nil, fmt.Errorf("parsing Maintenance Assignment Virtual Machine ID: (%q)", input)
```